packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1234,7 +1310,7 @@ export class MergeTree {
 			const deltaSegments: IMergeTreeSegmentDelta[] = [];
 			const overlappingRemoves: boolean[] = [];
 			pendingSegmentGroup.segments.map((pendingSegment: ISegmentLeaf) => {
-				const overlappingRemove = !pendingSegment.ack(pendingSegmentGroup, opArgs);
+				const overlappingRemove = !ackSegment(pendingSegment, pendingSegmentGroup, opArgs);
```

Comment:
i didn't do this initially on purpose, as there was/is active development in the area, and i'm trying to not clobber other people's work

Reply:
I am not sure what you are saying. This PR was made by squashing [test/remove-ISegment-deprecations](https://github.com/microsoft/FluidFramework/tree/test/remove-ISegment-deprecations) to main and then staging only the parts that should not be breaking. This was a change from your branch at the time - https://github.com/microsoft/FluidFramework/blob/802a21f85157d744af6cc0e879f15c72068cc39e/packages/dds/merge-tree/src/mergeTree.ts#L1300. The `ack` method is being removed so alternate call has to be made. `ackSegment` appears to be the refactored location - the refactor can be made now or later but yes any active development will be impacted. Recommendation is to do the refactor earlier so that the breaking changes are small and simple. Early movement helps prevent introduction of incompatible changes.